### DISCUSSION TITLE
Reading GOCAD Surfaces in GemGIS

### DISCUSSION THREAD

**Discussion by RichardScottOZ** [cgre-aachen/gemgis#175] [General]
This has saved me a bunch of time today, so thanks very much!

**Comment by RichardScottOZ** [comment]
Also, I imagine different versions of this software, e.g.

GOCAD TSurf 1 
HEADER {
*solid*color:0.694118 0.996078 0.796078
ivolmap:false
imap:false
*mesh:false
name:0490_BrokenHillGroup
}
GOCAD_ORIGINAL_COORDINATE_SYSTEM
NAME from_Shape
AXIS_NAME "U" "V" "W"
AXIS_UNIT "m" "m" "m"
ZPOSITIVE Elevation
END_ORIGINAL_COORDINATE_SYSTEM
TFACE
VRTX 1 1064644.9375 6413244.8203125 -2761.273681640625 
VRTX 2 1064643.375 6413214.7890625 -2768.81494140625 
VRTX 3 1064624.5 6412855.5625 -2500

**Comment by RichardScottOZ** [comment]
so the reader won't work looking for PVRTX or Properties

**Reply by AlexanderJuestel** [reply]
Hey, could you provide us with a simple gocad surface that has these properties? We set up the reader with surfaces that were available for us so we could just extend the function now :)

**Comment by RichardScottOZ** [comment]
Sure - see here:- https://github.com/pyvista/pyvista-support/issues/404

**Comment by RichardScottOZ** [comment]
you might be able to use some of the parsing hack I wrote that is at the end there.

**Comment by RichardScottOZ** [comment]
talked to Dom about it yesterday

**Comment by RichardScottOZ** [comment]
also http://paulbourke.net/dataformats/gocad/gocad.pdf

**Comment by RichardScottOZ** [comment]
https://github.com/pyvista/pyvista-support/issues/403

**Comment by AlexanderJuestel** [comment]
@RichardScottOZ alright, maybe we can do something during the Transform Hackathon :)

**Comment by RichardScottOZ** [comment]
Yes, please - my plan!

**Comment by RichardScottOZ** [comment]
GemGIS (maybe map2loop) - PyVista - SimPEG

**Comment by RichardScottOZ** [comment]
All the how can you pull orientations etc out of rasters/surfaces or whatever is lying around trick.

**Comment by RichardScottOZ** [comment]
Be good to have a display geological model, certainly.

**Comment by AlexanderJuestel** [comment]
The TS reader in GemGIS was updated in 1.0.1
